rollup/tracing/tracing.go

```diff
@@ -270,29 +288,44 @@ func (env *TraceEnv) getTxResult(state *state.StateDB, index int, block *types.B
 		}
 	}
 
-	tracer := vm.NewStructLogger(env.logConfig)
+	txContext := core.NewEVMTxContext(msg)
+	tracerContext := tracers.Context{
+		BlockHash: block.Hash(),
+		TxIndex:   index,
+		TxHash:    tx.Hash(),
+	}
+	callTracer, err := tracers.New("callTracer", &tracerContext)
+	if err != nil {
+		return fmt.Errorf("failed to create callTracer: %w", err)
+	}
+	prestateTracer, err := tracers.New("prestateTracer", &tracerContext)
+	if err != nil {
+		return fmt.Errorf("failed to create prestateTracer: %w", err)
+	}
+	structLogger := vm.NewStructLogger(env.logConfig)
+	tracer := NewMuxTracer(structLogger, callTracer, prestateTracer)
 	// Run the transaction with tracing enabled.
-	vmenv := vm.NewEVM(env.blockCtx, NewEVMTxContext(msg), state, env.chainConfig, vm.Config{Debug: true, Tracer: tracer, NoBaseFee: true})
+	vmenv := vm.NewEVM(env.blockCtx, txContext, state, env.chainConfig, vm.Config{Debug: true, Tracer: tracer, NoBaseFee: true})
 
 	// Call Prepare to clear out the statedb access list
 	state.Prepare(txctx.TxHash, txctx.TxIndex)
 
 	// Computes the new state by applying the given message.
 	l1DataFee, err := fees.CalculateL1DataFee(tx, state)
 	if err != nil {
-		return err
+		return fmt.Errorf("tracing failed: %w", err)
 	}
-	result, err := ApplyMessage(vmenv, msg, new(GasPool).AddGas(msg.Gas()), l1DataFee)
+	result, err := core.ApplyMessage(vmenv, msg, new(core.GasPool).AddGas(msg.Gas()), l1DataFee)
 	if err != nil {
-		return err
+		return fmt.Errorf("tracing failed: %w", err)
```

Comment:
TODO: fix behaviors

Reply:
fixed in https://github.com/scroll-tech/go-ethereum/pull/615/commits/ebd8a956f35ee9c82936844268a814be13ea553b